code/modules/clothing/suits/marine_armor.dm

```diff
@@ -171,6 +170,24 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	else armor_overlays["lamp"] = null
 	if(user) user.update_inv_wear_suit()
 
+
+/obj/item/clothing/suit/storage/marine/proc/post_creation_handling(mob/user)
+	if(!armor_variation)
+		return
+
+	if(!user || !user.client)
+		icon_state = replacetext(icon_state,"1","[rand(1,armor_variation)]")
+		update_icon()
+		return
+
+	if(user.client.prefs.preferred_armor)
+		var/new_look = armor_style_list.Find(user.client.prefs.preferred_armor)
+		if(new_look == 7)
```

Comment:
i dislike this being a magic number, a better way of doing all this would be making the global list a dict of {"style name": armor number}